### ISSUE TITLE
Type Alias Imported From Another File Not Respected

### ISSUE THREAD

**Issue by ivanthewebber** [microsoft/pyright#11305]
## Environment data

-   Pylance version: latest
-   OS and version: mac
-   Python version (& distribution if applicable, e.g. Anaconda): 3.14

## Code Snippet
```
from typing import Literal, TypeAlias
from enum import Enum, StrEnum

def enum_to_literal(enum_cls: Enum) -> TypeAlias:
    values = tuple(m.value for m in enum_cls)
    return Literal[values]

class A(StrEnum):
    a = "hi"

LiteralA = enum_to_literal(A)
```
```
from other import LiteralA

my_list: list[LiteralA] = list() # Pylance: Variable not allowed in type expression

<!--
Please provide a minimal, self-contained code snippet that reproduces the issue. If the code snippet uses any libraries, please specify the versions used.

Note: If you think a GIF of what is happening would be helpful, consider tools like https://github.com/vkohaupt/vokoscreenNG, https://www.cockos.com/licecap/, https://github.com/phw/peek or https://www.screentogif.com/ .
-->

```python
XXX
```

## Repro Steps

1. XXX

## Expected behavior

TypeHint accepted (w/wo explicit return type hint).

## Actual behavior

"Variable not allowed in type expression" 
Pylance[reportInvalidTypeForm](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportInvalidTypeForm.md)
On hover shows Any type in second file. Shows correct type (Literal or TypeHint) in first file.

## Logs

```
XXX
```

**Comment by erictraut** [comment]
Pyright's behavior here is correct. This isn't a bug. You're using the `TypeAlias` and `Literal` special forms incorrectly here. Refer to the [Python typing spec](https://typing.python.org/en/latest/spec/index.html) for details.

**Comment by ivanthewebber** [comment]
Yeah, sorry about that. I do wish enums came with a literal type hint, but I guess sometimes str must do.
